examples/components/CCFRAUD_VERTICAL/traininsilo/aml_comm.py

```diff
@@ -539,10 +590,19 @@ def send(self, data, destination) -> None:
 
         session_id = self._get_session_id(self._rank, destination)
         binary_data = pickle.dumps(data)
-        binary_data_size = sys.getsizeof(binary_data)
-        self._send(
-            math.ceil(binary_data_size / self._max_msg_size), session_id, destination
-        )
+        if self._encryption:
+            binary_data = self._encryption.encrypt(binary_data, destination)
+
+        binary_data_size = math.ceil(sys.getsizeof(binary_data) / self._max_msg_size)
```

Comment:
Oh this is because `binary_data_size` determines number of packets used to transfer `binary_data` as we would like to cap it at 256MB which is `self._max_msg_size`

Reply:
Thanks for the clarification. I was misled by the variable name. It is not really a size then, just a number of packets. Got it.

Follow-up question: isn't that this number that you should be using as the upper limit of the For loop in line 606 then?

Reply:
yes, thanks for noticing - I have fixed both the name and iterations